### ISSUE TITLE
"failed to parse" when parsing correctly

### ISSUE THREAD

**Issue by vikaslalwani** [suzuki-shunsuke/tfcmt#1547]
### tfcmt version

```console
$ tfcmt -v
tfcmt version 4.2.0 (ca834049d0655ab9eec79e913aa7eacc362c2f9e)
```
v4.2.0

### Environment

* OS: Linux
* CPU Architecture: amd64

### Overview

It says "It failed to parse the result", when it actually did parse the results correctly. It happens only when there are  no changes to actual plan but only outputs are there as a result of plan.

![image](https://github.com/user-attachments/assets/8cd19589-e365-4ffc-bc05-fbc96bf78b7c)

### How to reproduce

tfcmt.yaml

<details> <summary>tfcmt.yaml</summary>
<p>

```yaml
# Based upon: https://suzuki-shunsuke.github.io/tfcmt/config#default-configuration
embedded_var_names: []
ci:
  pr: []
  owner: []
  repo: []
  sha: []
  link: []
  vars: {}
templates:
  plan_title: "## {{if eq .ExitCode 1}}:x: {{end}}Plan Result{{if .Vars.target}} ({{.Vars.target}}){{end}}"
  apply_title: "## :{{if eq .ExitCode 0}}white_check_mark{{else}}x{{end}}: Apply Result{{if .Vars.target}} ({{.Vars.target}}){{end}}"

  result: |
    {{if .Vars.invocation }}> `{{.Vars.invocation}}` {{end}}
    {{if .Vars.posted_by }}> `{{.Vars.posted_by}}` {{end}}
    {{if .Result}}<pre><code>{{ .Result }}</code></pre>{{end}}
  updated_resources: |
    {{if .CreatedResources}}
    * Create
    {{- range .CreatedResources}}
      * {{.}}
    {{- end}}{{end}}{{if .UpdatedResources}}
    * Update
    {{- range .UpdatedResources}}
      * {{.}}
    {{- end}}{{end}}{{if .DeletedResources}}
    * Delete
    {{- range .DeletedResources}}
      * {{.}}
    {{- end}}{{end}}{{if .ReplacedResources}}
    * Replace
    {{- range .ReplacedResources}}
      * {{.}}
    {{- end}}{{end}}
  deletion_warning: |
    {{if .HasDestroy}}
    ### :warning: Resource Deletion will happen :warning:
    This plan contains resource delete operation. Please check the plan result very carefully!
    {{end}}
  changed_result: |
    {{if .ChangedResult}}
    <details><summary>Change Result (Click me)</summary>
    {{wrapCode .ChangedResult}}
    </details>
    {{end}}
  change_outside_terraform: |
    {{if .ChangeOutsideTerraform}}
    <details><summary>:information_source: Objects have changed outside of Terraform</summary>

    _This feature was introduced from [Terraform v0.15.4](https://github.com/hashicorp/terraform/releases/tag/v0.15.4)._
    {{wrapCode .ChangeOutsideTerraform}}
    </details>
    {{end}}
  warning: |
    {{if .Warning}}
    ## :warning: Warnings :warning:
    {{wrapCode .Warning}}
    {{end}}
  error_messages: |
    {{if .ErrorMessages}}
    ## :warning: Errors
    {{range .ErrorMessages}}
    * {{. -}}
    {{- end}}{{end}}
  guide_apply_failure: ""
  guide_apply_parse_error: ""
terraform:
  plan:
    disable_label: false
    template: |
      {{template "plan_title" .}}

      {{if .Link}}[CI link]({{.Link}}){{end}}

      {{template "deletion_warning" .}}
      {{template "result" .}}
      {{template "updated_resources" .}}

      {{template "changed_result" .}}
      {{template "change_outside_terraform" .}}
      {{template "warning" .}}
      {{template "error_messages" .}}
    when_add_or_update_only:
      label: "{{if .Vars.target}}{{.Vars.target}}/{{end}}add-or-update"
      label_color: 1d76db # blue
    when_destroy:
      label: "{{if .Vars.target}}{{.Vars.target}}/{{end}}destroy"
      label_color: d93f0b # red
    when_no_changes:
      label: "{{if .Vars.target}}{{.Vars.target}}/{{end}}no-changes"
      label_color: 0e8a16 # green
    when_plan_error:
      label:
      label_color:
    when_parse_error:
      template: |
        {{template "plan_title" .}}

        {{if .Link}}[CI link]({{.Link}}){{end}}

        It failed to parse the result.

        <details><summary>Details (Click me)</summary>
        {{wrapCode .CombinedOutput}}
        </details>
  apply:
    template: |
      {{template "apply_title" .}}

      {{if .Link}}[CI link]({{.Link}}){{end}}

      {{if ne .ExitCode 0}}{{template "guide_apply_failure" .}}{{end}}

      {{template "result" .}}

      <details><summary>Details (Click me)</summary>
      {{wrapCode .CombinedOutput}}
      </details>
      {{template "error_messages" .}}`
    when_parse_error:
      template: |
        {{template "apply_title" .}}

        {{if .Link}}[CI link]({{.Link}}){{end}}

        {{template "guide_apply_parse_error" .}}

        It failed to parse the result.

        <details><summary>Details (Click me)</summary>
        {{wrapCode .CombinedOutput}}
        </details>
```

</p>
</details>

Terraform Configuration

* Not relevant. 

Scenario - if there are no changes to the actual plan but only outputs are being dumped it goes to `when_parse_error` section but it's actually parsing it right.

Executed command and output

```console
$ terraform plan -var-file=myvars.tfvars -out tfplan.binary -input=false

$ terraform show tfplan.binary > tmp_plan.out

$ tfcmt --config "$HOME"/tfcmt.config.yaml \
        --owner myorg --repo myrepo --pr myPrNumber \
        --var invocation:"myinvocation" --var posted_by:"myuser" \
        --var target:"mytarget" plan --patch -- cat tmp_plan.out
```

### Expected behaviour

The result should not be dumped from `when_parse_error` but from `when_no_changes`

### Actual behaviour

Goes to `when_parse_error` when actually parsing correctly

**Comment by suzuki-shunsuke** [comment]
I can't reproduce the issue with a simple case.

main.tf

```tf
output "foo" {
  value = "foo"
}
```

```console
$ tfcmt -v
tfcmt version 4.14.0 (13b3b64b1444d528db49d60a99310bcd45993a52)
```

```sh
terraform init
```

```console
$ tfcmt plan -- terraform plan

Changes to Outputs:
  + foo = "foo"

You can apply this plan to save these new output values to the Terraform
state, without changing any real infrastructure.

─────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't
guarantee to take exactly these actions if you run "terraform apply" now.

```

<img width="942" alt="Image" src="https://github.com/user-attachments/assets/ec404ae4-0ee5-4189-b8f4-19770a17cfe6" />

**Comment by suzuki-shunsuke** [comment]
> v4.2.0

v4.2.0 is old.

https://github.com/suzuki-shunsuke/tfcmt/releases/tag/v4.2.0

Could you try the latest version?

**Comment by suzuki-shunsuke** [comment]
Okay.
I've confirmed this issue was resolved at v4.5.1.

- https://github.com/suzuki-shunsuke/tfcmt/pull/908

I could reproduce the issue with tfcmt v4.5.0, but couldn't reproduce with tfcmt v4.5.1.
